chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -382,198 +418,207 @@
-                target_platform_version: target_platform_version)
+    # Make sure we get a crowbar install key as well.
+    unless crowbar_key.empty?
+      append << " crowbar.install.key=#{crowbar_key}"
     end
 
-    missing_files = ! File.exists?("#{os_dir}/install/boot/x86_64/common")
+    # These should really be made libraries or something.
+    case
+    when /^(suse)/ =~ os
+      # Add base OS install repo for suse
+      node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install" => true }
+
+      ntp_servers = search(:node, "roles:ntp-server")
+      ntp_servers_ips = ntp_servers.map { |n| Chef::Recipe::Barclamp::Inventory.get_network_by_type(n, "admin").address }
+
+      target_platform_version = os.gsub(/^.*-/, "")
+      template "#{os_dir}/crowbar_join.sh" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "crowbar_join.suse.sh.erb"
+        variables(admin_ip: admin_ip,
+                  web_port: web_port,
+                  ntp_servers_ips: ntp_servers_ips,
+                  target_platform_version: target_platform_version)
+      end
 
-  when /^(redhat|centos)/ =~ os
-    # Add base OS install repo for redhat/centos
-    if ::File.exists? "#{tftproot}/#{os}/install/repodata"
-      node.set[:provisioner][:repositories][os]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install" => true }
-    else
-      node.set[:provisioner][:repositories][os]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install/Server" => true }
-    end
-    # Default kickstarts and crowbar_join scripts for redhat.
+      Provisioner::Repositories.inspect_repos(node)
+      repos = Provisioner::Repositories.get_repos(node, "suse", target_platform_version, arch)
+
+      packages = node[:provisioner][:packages][os] || []
+
+      template "#{os_dir}/crowbar_register" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "crowbar_register.erb"
+        variables(admin_ip: admin_ip,
+                  admin_broadcast: admin_broadcast,
+                  web_port: web_port,
+                  ntp_servers_ips: ntp_servers_ips,
+                  os: os,
+                  arch: arch,
+                  crowbar_key: crowbar_key,
+                  domain: domain_name,
+                  repos: repos,
+                  packages: packages,
+                  target_platform_version: target_platform_version)
+      end
 
-    template "#{os_dir}/crowbar_join.sh" do
-      mode 0644
-      owner "root"
-      group "root"
-      source "crowbar_join.redhat.sh.erb"
-      variables(admin_web: admin_web,
-                os_codename: os_codename,
-                crowbar_repo_web: crowbar_repo_web,
-                admin_ip: admin_ip,
-                provisioner_web: provisioner_web,
-                web_path: web_path)
-    end
+      missing_files = !File.exists?("#{os_dir}/install/boot/#{arch}/common")
```

Comment:
Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.